Library/Homebrew/prefix.sh

```diff
@@ -25,17 +25,33 @@ homebrew-prefix() {
   [[ -z "${prefix}" ]] && return 1
   [[ -z "${formula}" ]] && echo "${HOMEBREW_PREFIX}" && return 0
 
-  local formula_path
+  local formula_exists
   if [[ -f "${HOMEBREW_REPOSITORY}/Library/Taps/homebrew/homebrew-core/Formula/${formula}.rb" ]]
   then
-    formula_path="${HOMEBREW_REPOSITORY}/Library/Taps/homebrew/homebrew-core/Formula/${formula}.rb"
+    formula_exists="1"
   else
+    local formula_path
     formula_path="$(
       shopt -s nullglob
       echo "${HOMEBREW_REPOSITORY}/Library/Taps"/*/*/{Formula/,HomebrewFormula/,}"${formula}.rb"
     )"
+    [[ -n "${formula_path}" ]] && formula_exists="1"
   fi
-  [[ -z "${formula_path}" ]] && return 1
+
+  if [[ -z "${formula_exists}" &&
+        -z "${HOMEBREW_NO_INSTALL_FROM_API}" &&
+        -f "${HOMEBREW_CACHE}/api/formula.json" ]]
+  then
+    formula_exists="$(
+      ruby -rjson <<RUBY 2>/dev/null
```

Comment:
Not sure we're guaranteed a `ruby` here, are we? Not a dealbreaker though, since I suppose we can still fallback to `--prefix.rb`.

Reply:
But, if so (or even alternatively to avoid having to fire up `ruby`), perhaps we can consider doing
```sh
[[ -h "${HOMEBREW_PREFIX}/opt/${formula}" ]]
```
somewhere earlier to handle the case where the formula is installed.

Reply:
We're not, though that but would also affect `brew formulae`. And yes, if it's not present it will fall back.

Maybe could do what `_create_lock` does and fallback to Python and then something else.

In fact the consequences are worse for `brew formulae` as it breaks shell completion etc so it may be worth improving that.

Reply:
Or maybe just hardcode a PATH to portable-ruby vendor directory somewhere. System Ruby 2.6 outside of macOS is pretty much dead at this point so there's usually a portable Ruby installed.

Reply:
> But, if so (or even alternatively to avoid having to fire up `ruby`), perhaps we can consider doing
> 
> ```shell
> [[ -h "${HOMEBREW_PREFIX}/opt/${formula}" ]]
> ```
> 
> somewhere earlier to handle the case where the formula is installed.

Oops missed this somehow. Decent idea actually. The only difference is the deleted formulae case, which could be argued as an improvement.

Reply:
> We're not, though that but would also affect `brew formulae`.

I think it'd be nice to have a shared `.sh` file with a function that both of these can use to do this JSON operation?

> Or maybe just hardcode a PATH to portable-ruby vendor directory somewhere.

Let's not unless we get a bug report.